Sources/DistributedCluster/LifecycleMonitoring/LifecycleWatchContainer.swift

```diff
@@ -50,10 +50,10 @@ final class LifecycleWatchContainer {
     }
 
     func clear() {
-//        _lock.wait()
-//        defer {
-//            _lock.signal()
-//        }
+        self._lock.wait()
+        defer {
+            _lock.signal()
+        }
 
         traceLog_DeathWatch("Clear LifecycleWatchContainer owned by \(self.watcherID)")
         self.watching = [:]
```

Comment:
unrelated: is the `FIXME` on L61 still relevant?

Reply:
I should probably revisit it right away; I think my suspicion there was about if we needed to remove the self watcher ID as well but didn't -- I'll check this during this PR